src/common/summarize-profile.js

```diff
@@ -110,17 +97,43 @@ export function summarizeProfile(profile: Profile) {
  */
 function functionNameCategorizer() {
   const cache = new Map();
+
+  // Compile set of categories into a single regular expression that produces
+  // the pattern (whether it's a prefix, stem, substring, or exact match).
+
+  const patternToCategory = new Map();
+
+  const matchPatterns = {
+    prefix: [],
+    stem: [],
+    substring: [],
+    exact: []
+  };
+
+  for (const [matchType, pattern, category] of categories) {
+    patternToCategory.set(pattern, category);
+    let pat = pattern.replace(/\W/g, '\\$&');
+    if (matchType == 'stem')
+      pat += "\\b(?=\\()?";
+    matchPatterns[matchType].push(pat);
+  }
+
+  const regex = new RegExp(  "^(?:"    + matchPatterns['prefix'].join("|") + ")"
+                           + "|(?:"    + matchPatterns['substring'].join("|") + ")"
+                           + "|\\b(?:" + matchPatterns['stem'].join("|") + ")"
+                           + "|^(?:"   + matchPatterns['exact'].join("|") + ")$");
```

Comment:
It might be ok, but it would require a little more maintenance -- return types tend to change (bool <-> void is common), namespaces occasionally change.

But really, most of this table should be replaced by pseudostack frames with categories. Unfortunately, we still haven't unified the two notions of categories. (Mostly my fault, I think.)